### PR TITLE
Chore/add elm example

### PR DESCRIPTION
**Pull request by ChristophP**
Woot woot running elm example.
All you need to do is run `npm run build` to create the build in the dist folder. Then simply open `examples/elm/index.html` in the browser form the file system.

Tricky parts to get it running were:
-> getting the original React event from the custom event that is being fired
-> really wrapping my head around it if something is not working(web components are really not very intuitive for me yet, working with the DOM so much kinda sucks :-) )

Anyway it's working which is awesome. make sure to check the browser console for some event that are logged.

One issue:
When loading the page there are two errors on the console. These are probably due to some timing issues between react and elm. Didn't have time to look into that yet but it works anyway for now.

### CODE REVIEW

File: examples/elm/Main.elm [hunk 1]
```diff
@@ -53,38 +58,51 @@ update msg model =
...
                 [ id "customComponent"
-                , attribute "name" model.name
-                , attribute "value" model.value
-                , property "disabled" (Encode.bool model.disabled)
+                , name model.name
+                , value model.value
+                , disabled model.disabled
+                , hasError model.hasError
                 , attribute "options"
                     "[{ \"value\": \"one\", \"label\": \"One\" },{ \"value\": \"two\", \"label\": \"Two\" }]"
```
Comment (layflags) [inline examples/elm/Main.elm]: We could also use smth like this, no?
```elm
property "options" << Encode.list 
    [ object [ ("value", string "one"), ("label", string "One") ]
    , object [ ("value", string "two"), ("label", string "Two") ]
    ] 
``` 
Or provide a custom `options` function that expects a custom type like `List Option`.
```
options [ Option "one" "One", Option "two" "Two" ]
```

File: examples/elm/Main.elm [hunk 2]
```diff
@@ -53,38 +58,51 @@ update msg model =
...
         [ h3 [] [ text "Custom select component" ]
         , div []
             [ customSelect
                 [ id "customComponent"
-                , attribute "name" model.name
-                , attribute "value" model.value
-                , property "disabled" (Encode.bool model.disabled)
+                , name model.name
+                , value model.value
+                , disabled model.disabled
```
Comment (layflags) [inline examples/elm/Main.elm]: nice when using the predefined functions :)

File: examples/elm/Main.elm [hunk 3]
```diff
@@ -53,38 +58,51 @@ update msg model =
...
+                , hasError model.hasError
```
Comment (layflags) [inline examples/elm/Main.elm]: Would be nice to have a separate module for `customSelect` with all related helpers and stuff in.

File: examples/elm/Main.elm [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+view : Model -> Html Msg
+view model =
+    div []
+        [ h3 [] [ text "Custom select component" ]
+        , div []
+            [ customSelect
+                [ id "customComponent"
+                , name model.name
+                , value model.value
```
Comment (layflags) [inline examples/elm/Main.elm]: attributes `name` and `value` are passed JSON encoded (with escaped double quotes)
screen shot 2017-06-23 at 23 43 12


doesn't happen with: 
```elm
                , attribute "name" model.name
                , attribute "value" model.value
```

File: examples/elm/Main.elm (hunk 3 above)
Comment (ChristophP) [inline examples/elm/Main.elm]: Hmm, I thought leaving it in one file will keep the example as easy as possible. But good idea for production. Do you still think I should move it?

File: examples/elm/Main.elm (hunk 4 above)
Comment (ChristophP) [inline examples/elm/Main.elm]: Oh good catch!

File: examples/elm/Main.elm (hunk 1 above)
Comment (ChristophP) [inline examples/elm/Main.elm]: Yeah sure I will add that.

File: examples/elm/Main.elm (hunk 3 above)
Comment (layflags) [inline examples/elm/Main.elm]: Yes sure, for this example everything should be in one file.

**Comment by ChristophP** [comment]
you still have to check the last commit.

**Pull request merged**